contracts/core/extensions/CoreIssuance.sol

```diff
@@ -115,6 +116,75 @@ contract CoreIssuance is
         }
     }
 
+    /**
+     * Composite method to redeem and withdraw with a single transaction
+     *
+     * Normally, you should expect to be able to withdraw all of the tokens.
+     * However, some have central abilities to freeze transfers (e.g. EOS). _toWithdraw
+     * allows you to optionally specify which component tokens to transfer
+     * back to the user. The rest will remain in the vault under the users' addresses.
+     *
+     * @param _setAddress   The address of the Set token
+     * @param _quantity     The number of tokens to redeem
+     * @param _toWithdraw   Mask of indexes of tokens to withdraw
+     */
+    function redeemAndWithdraw(
+        address _setAddress,
+        uint _quantity,
+        uint _toWithdraw
+    )
+        external
+        isValidSet(_setAddress)
+        isPositiveQuantity(_quantity)
+        isNaturalUnitMultiple(_quantity, _setAddress)
+    {
+        // Burn the Set token (thereby decrementing the SetToken balance)
+        ISetToken(_setAddress).burn(msg.sender, _quantity);
+
+        // Fetch Set token properties
+        uint naturalUnit = ISetToken(_setAddress).naturalUnit();
+        address[] memory components = ISetToken(_setAddress).getComponents();
+        uint[] memory units = ISetToken(_setAddress).getUnits();
+
+        // Loop through and decrement vault balances for the set, withdrawing if requested
+        for (uint i = 0; i < components.length; i++) {
+
+            // Calculate quantity to transfer
+            uint componentQuantity = calculateTransferValue(
+                 units[i],
+                naturalUnit,
+                _quantity
+            );
+
+            // Decrement the Set amount
```

Comment:
we're decrementing the `component` amount right? not the `Set` amount

Reply:
+1